src/assignments/tech-jobs-oo.rst

```diff
@@ -0,0 +1,580 @@
+-----------------------------------
+
+Review the code in the ``Employer``, ``Location``, ``CoreCompetency``, and
+``PositionType`` classes. What similarities do you see?
+
+There is a fair amount of repetition between the classes. As a good coder,
+anytime you find yourself adding identical code in multiple locations you
+should consider how to streamline the process.
+
+   DRY = "Don't Repeat Yourself".
+
+Create a Base Class
+^^^^^^^^^^^^^^^^^^^
+
+Let's move all of the repeated code into a separate class. We will then have
+``Employer``, ``Location``, ``CoreCompetency``, and ``PositionType`` *inherit*
+this common code.
+
+#. Create a new class called ``JobField``.
+#. Consider the following questions to help you decide what code to put in the
+   ``JobField`` class:
+
+   a. What fields do ALL FOUR of the classes have in common?
+   b. Which constructors are the same in ALL FOUR classes?
+   c. Which custom methods are identical in ALL of the classes?
+
+#. In ``JobField``, declare each of the common fields.
+#. Code the constructors.
+#. Add in the custom methods.
+#. Finally, to prevent the creation of a ``JobField`` object, make this class
+   *abstract*.
+
+Extend ``JobField`` into ``Employer``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Now that you have the common code located in the ``JobField`` file, we can
+modify the other classes to reference this shared code. Let's begin with
+``Employer``.
+
+#. Modify line 5 to *extend* the ``JobField`` class into ``Employer``.
+
+   .. sourcecode:: csharp
+      :lineno-start: 5
+
+      public class Employer : JobField
+      {
+         //Code not displayed.
+      }
+
+#. Next, remove any code in ``Employer`` that matches code from ``JobField``
+   (e.g. the ``Id``, ``Value``, and ``nextId`` fields are shared).
+#. Remove any of the custom methods that are identical.
+#. The empty constructor is shared, but not the second. Replace the two
+   constructors with the following:
+
+   .. sourcecode:: csharp
+      :lineno-start: 7
+
+      public Employer(string value) : base() 
+      {
+         Value = value;
+      }
+
+   The ``:`` and ``base`` keywords link the ``JobField`` and
+   ``Employer`` classes.
+#. Rerun your unit tests to verify your refactored code.
+
+Finish DRYing Your Code
+^^^^^^^^^^^^^^^^^^^^^^^
+
+#. Repeat the process above for the ``Location``, ``CoreCompetency``, and
+   ``PositionType`` classes.
+#. Rerun your unit tests to verify that your classes and methods still work.
+
+.. admonition:: Tip
+
+   You know you need to do this, but here is the reminder anyway. Save, commit,
+   and push your work to GitHub.
+
+Sanity Check
+-------------
+
+Once you finish all of the tasks outlined above, all that remains is to check
+the console display.
+
+Sally has provided some commented-out code in ``Main`` that prints out a small
+list of ``Job`` objects. Go ahead and activate this code and run it.
+Properly done, your output should look something like:
```

Comment:
fyi this is not in the current version of the starter code - also something we talked about with that studio repo, add a gitignore so we're not trekking around a vs settings file. 
also,  you may wnat to clarify you're referring to the main method in program, unless the aim is to have the students deduce for themselves that you are referring to a method in a class file and not a class name.

Reply:
Fixed!